### DISCUSSION TITLE
Is there a way to ignore rendering air blocks?

### DISCUSSION THREAD

**Discussion by benhovinga** [maruohon/litematica#922] [Q&A] [answered]
I have a schematic that is to be placed underground. It is mostly air but when looking at it using a mod like Freecam, all I see is a large pink box. Is there a way to disable (or ignore) rendering air blocks in the schematic? This way I can focus on digging out the small area required for the build and not the entire footprint of the schematic.

The below image is of the schematic in it's intended place and viewed with Freecam underground.

![2024-12-13_19 28 28](https://github.com/user-attachments/assets/3e60569b-abed-4ac4-84d1-be9a6daea40a)

The next image is of the schematic moved above the ground surface to see how much is just air.

![2024-12-13_19 30 36](https://github.com/user-attachments/assets/959d3f77-18f9-4431-a8e8-d94c66cbbbbe)

> This is my custom Iron/Crop farm and wishing well design I've used in a few worlds.

**Comment by maruohon** [comment, marked as answer]
In the Visuals category, disable ´schematicOverlayTypeExtra`, `schematicOverlayTypeMissing` and `enableSchematicBlocksRendering`. Then you are left with the wrong block and wrong state red and orange overlay around the blocks to dig out.

Keeping the extra type overlay enabled is also an option, as once you start digging out then it won't be at your face as it will disappear from all the already dug out air blocks.

**Reply by benhovinga** [reply]
Thanks that worked. I only disabled `schematicOverlayTypeExtra` and `schematicOverlayTypeMissing`. I left `enableSchematicBlocksRendering` enabled so I can see the blocks and not just wireframe.

![2024-12-13_20 12 12](https://github.com/user-attachments/assets/f91af49c-14ba-4518-9c0b-f0bac39ea77f)

I'll get around to learning sub-regions eventually, but currently prefer to quickly select the whole structure and just saving it for speed (and laziness).

**Comment by maruohon** [comment]
Also with a schematic shaped like this, I would recommend using multiple sub-regions to more tightly capture the parts of the build and avoid the large areas of just air. There is an article in the wiki here about the "Normal"/Multi-Region area selection mode if you are unfamiliar with it.
